### PR TITLE
Add tests for accessor names

### PR DESCRIPTION
**Pull request by jugglinmike**
These tests for accessor names were originally submitted in gh-706, but that
change set only concerned their use within object initializers. At @avdg's
recommendation, I have expanded coverage to include class definitions. Because
this generally concerns four additional syntactic forms (instance accessors
within class expressions, static accessors within class expressions, instance
accessors within class declarations, and static accessors within class
declarations), I've made use of the test generation tool to promote consistent
coverage. Not all tests were suitable subjects for procedural generation,
however (there is only one interesting case for the use of the `in` keyword,
for example, making the "case/template" formatting needlessly indirect), so
this patch also includes some traditional non-generated test material. In all
cases, file meta-data and license information accurately describes the
provenance.

### CODE REVIEW

File: src/accessor-names/literal-string-double-quote.case [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  ComputedPropertyName : [ AssignmentExpression ]
+
+  1. Let exprValue be the result of evaluating AssignmentExpression.
+  2. Let propName be ? GetValue(exprValue).
+  3. Return ? ToPropertyKey(propName).
+---*/
+
+//- declareWith
+"singleQuote"
```
Comment (gnarf) [inline src/accessor-names/literal-string-double-quote.case]: Is this just a beautiful troll? or did you mean `"doubleQuote"`

File: test/language/expressions/class/accessor-name-static-computed-yield-expr.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  2. Let propName be ? GetValue(exprValue).
+  3. Return ? ToPropertyKey(propName).
+features: [generators]
+---*/
+
+var yieldSet, C, iter;
+function* g() {
+  C = class {
+    static get [yield]() { return 'get yield'; }
+    static set [yield](param) { yieldSet = param; }
```
Comment (gnarf) [inline test/language/expressions/class/accessor-name-static-computed-yield-expr.js]: I want to like frame this one in the next sideshow blogpost...

**Comment by gnarf** [comment]
Overall 👍  they all pass locally on the node harness even!

File: test/language/expressions/class/accessor-name-static-computed-yield-expr.js (hunk 2 above)
Comment (jugglinmike) [inline test/language/expressions/class/accessor-name-static-computed-yield-expr.js]: :)

Comment (jugglinmike) [inline test/language/expressions/class/accessor-name-static-computed-yield-expr.js]: :)

File: src/accessor-names/literal-string-double-quote.case (hunk 1 above)
Comment (jugglinmike) [inline src/accessor-names/literal-string-double-quote.case]: Yup, that was an oversight on my part. Fixed

**Comment by jugglinmike** [comment]
Thanks for the review, Corey!

**Comment by littledan** [comment]
+1 happy to have this extra test coverage; as @bakkot can tell you, the cases related to contextual keywords and will only get worse with future proposals; @anba has found bugs in V8 related to using yield in weird places like this. So this is all great to have these systematic tests.

**Comment by leobalter** [comment]
LGTM.

Merged at 228851fbe and 37d5f8e5e

**Comment by leobalter** [comment]
I really loved using `yield` to create dynamic accessor names in classes. It should be the new factory pattern.

**Pull request closed without merging**